File: p2p_library/src/main/java/com/tambapps/p2p/fandem/listener/SendingListener.java
package com.tambapps.p2p.fandem.listener;

import com.tambapps.p2p.fandem.Peer;

public interface SendingListener extends TransferListener {

  /**
   * called when the sender started the sending service
   * @param self the self peer
   * @param fileName the name of the file that will be sent
   */
  void onStart(Peer self, String fileName);

}

File: p2p_library/src/main/kotlin/com/tambapps/p2p/fandem/listener/SendingListener.kt
package com.tambapps.p2p.fandem.listener

import com.tambapps.p2p.fandem.Peer

interface SendingListener : TransferListener {
    /**
     * called when the sender started the sending service
     * @param self the self peer
     * @param fileName the name of the file that will be sent
     */
    fun onStart(self: Peer, fileName: String)
}